In [1]:
import pandas as pd
import numpy as np
from IPython.display import Image
import matplotlib.pyplot as plt
import seaborn as sns

import os
for dirname, _, filenames in os.walk('Data'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

Data\Olympics_Country.csv
Data\Olympics_Games.csv
Data\Olympic_Athlete_Bio.csv
Data\Olympic_Athlete_Event_Results.csv
Data\Olympic_Results.csv
Data\Pipfile


## Data Exploration

### Athlete Bio Data

##### Primary Key => athleteID

In [2]:
path_Olympic_Athlete_BIO ="Data\Olympic_Athlete_BIO.csv"
Athlete_Bio = pd.read_csv(path_Olympic_Athlete_BIO, sep = ",")
Athlete_Bio.head()

athlete_id                 name     sex        born height weight  \
0       16809       Károly Teppert    Male  1891-07-20     na     na   
1       43737    Andrzej Socharski    Male  1947-08-31  173.0     72   
2       50147  Nathalie Wunderlich  Female  1971-06-03  170.0     50   
3        5085           Miha Lokar    Male  1935-09-10  182.0     76   
4      136329          Austin Hack    Male  1992-05-17  203.0    100   

          country country_noc  \
0         Hungary         HUN   
1          Poland         POL   
2     Switzerland         SUI   
3      Yugoslavia         YUG   
4   United States         USA   

                                         description  \
0  Károly Teppert started competing in cycling in...   
1                                                 na   
2                                                 na   
3                                                 na   
4                                                 na   

                                       special_notes  
0                                                 na  
1  Listed in Olympians Who Won a Medal at the Wor...  
2                                                 na  
3  Listed in Olympians Who Won a Medal at the Eur...  
4                                                 na

In [3]:
# Drop Description and Special Notes as they will not be necessary
AB = Athlete_Bio.drop(columns = ['description','special_notes'])
AB.head()

athlete_id                 name     sex        born height weight  \
0       16809       Károly Teppert    Male  1891-07-20     na     na   
1       43737    Andrzej Socharski    Male  1947-08-31  173.0     72   
2       50147  Nathalie Wunderlich  Female  1971-06-03  170.0     50   
3        5085           Miha Lokar    Male  1935-09-10  182.0     76   
4      136329          Austin Hack    Male  1992-05-17  203.0    100   

          country country_noc  
0         Hungary         HUN  
1          Poland         POL  
2     Switzerland         SUI  
3      Yugoslavia         YUG  
4   United States         USA

#### Missing Values

In [4]:
AB.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 155031 entries, 0 to 155030
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   athlete_id   155031 non-null  int64 
 1   name         155031 non-null  object
 2   sex          155031 non-null  object
 3   born         155031 non-null  object
 4   height       155031 non-null  object
 5   weight       155031 non-null  object
 6   country      155031 non-null  object
 7   country_noc  155031 non-null  object
dtypes: int64(1), object(7)
memory usage: 9.5+ MB


#### Duplicate values

In [5]:
AB.shape

(155031, 8)

In [6]:
AB['name'].unique().shape

(151955,)

As we can see in the above count of the rows, there is a difference in the count, which may suggest that there may be duplicate names in the data. This however may not be the case as the athlete id has distinct values.

#### QS: Do we drop the duplicate name values?

#### Not available as value, Height and Weight

In [7]:
not_availableAB = AB.loc[np.logical_and(AB['height'] == 'na', AB['weight'] == 'na')]
not_availableAB

athlete_id                        name     sex        born height  \
0            16809              Károly Teppert    Male  1891-07-20     na   
6            24791                 Saad Rashed    Male  1928-11-20     na   
9            74076            Hassan Illiassou    Male  1966-09-22     na   
12           35720               Achille Mengé    Male          na     na   
14          142137      Nanna Skodborg Merrald  Female  1993-10-08     na   
...            ...                         ...     ...         ...    ...   
155013       19515               Jo Myeong-Jun    Male  1970-07-29     na   
155014       25572      Ioannis Andrianopoulos    Male  1900-01-01     na   
155018       51182             Frances Schroth  Female  1893-04-11     na   
155019      146786         Gabriel Gerald Geay    Male  1996-09-10     na   
155027      701697  Hendrik Mangelaar Meertens    Male  1908-09-30     na   

       weight                                         country country_noc  
0          na                                         Hungary         HUN  
6          na                                           Egypt         EGY  
9          na                                           Niger         NIG  
12         na                                         Belgium         BEL  
14         na                                         Denmark         DEN  
...       ...                                             ...         ...  
155013     na   People's Republic of China  Republic of Korea         CHN  
155014     na                                          Greece         GRE  
155018     na                                   United States         USA  
155019     na                     United Republic of Tanzania         TAN  
155027     na                                     Netherlands         NED  

[50547 rows x 8 columns]

The data has 50547 not available number for weight and height. This shows that a third of the data is missing for the height and weight.
Due to lack of a business case, we still  have to estimate the necessity of these two columns.


There is no need to look for an outlier as this data represents the BIO of a player, outliers cannot take effect in this data.
Medals per country per sport per per addition of olympics per athleter per gender

#### Ratio Gender

In [8]:
AB["sex"].value_counts()

Male      114755
Female     40276
Name: sex, dtype: int64